core/src/main/java/jenkins/install/InstallUtil.java

```diff
@@ -193,7 +188,7 @@ private static InstallState getDefaultInstallState() {
             // Edge case: used Jenkins 1 but did not save the system config page,
             // the version is not persisted and returns 1.0, so try to check if
             // they actually did anything
-            if (!j.getItemMap().isEmpty() || !mayBeJenkins2SecurityDefaults(j) || !j.getNodes().isEmpty()) {
```

Comment:
We cannot rely on security settings for determining if this is an upgrade or new install; scripts may set up security